### ISSUE TITLE
Evaluation exceeded 1000 backwards branches when printing kernel32.GetLastError

### ISSUE THREAD

**Issue by marler8997** [ziglang/zig#5757]
Reproduce on windows with:
```zig
const std = @import("std");
pub fn main() void {
    std.debug.warn("{}\n", .{std.os.windows.kernel32.GetLastError()});
}
```

Output on my machine:
```
D:\tools\zig-windows-x86_64-0.6.0+e120b07a5\lib\zig\std\fmt.zig:369:20: error: evaluation exceeded 1000 backwards branches
            inline for (enumInfo.fields) |enumField| {
                   ^
D:\tools\zig-windows-x86_64-0.6.0+e120b07a5\lib\zig\std\fmt.zig:182:35: note: called from here
                    try formatType(
                                  ^
D:\tools\zig-windows-x86_64-0.6.0+e120b07a5\lib\zig\std\io\writer.zig:28:34: note: called from here
            return std.fmt.format(self, format, args);
                                 ^
D:\tools\zig-windows-x86_64-0.6.0+e120b07a5\lib\zig\std\debug.zig:65:27: note: called from here
    nosuspend stderr.print(fmt, args) catch return;
                          ^
.\src\bug.zig:3:19: note: called from here
    std.debug.warn("{}\n", .{std.os.windows.kernel32.GetLastError()});
                  ^
.\src\bug.zig:2:20: note: called from here
pub fn main() void {
                   ^
D:\tools\zig-windows-x86_64-0.6.0+e120b07a5\lib\zig\std\fmt.zig:182:35: note: referenced here
                    try formatType(
                                  ^

```

**Comment by marler8997** [comment]
potential fix here: https://github.com/ziglang/zig/pull/5773

**Comment by andrewrk** [comment]
Fixed by @marler8997 in #5773, landed in 27c1e0b453d50f8c242fd1153290a00b74dff018.
